### PR TITLE
apps: mv kubernetes-external-secrets and node-perf-dash, update existing, add README

### PR DESCRIPTION
**Pull request by spiffxp**
Part of umbrella issue https://github.com/kubernetes/k8s.io/issues/2150

I spiraled around the existing apps migrated (gcsweb, sippy) and moved over two more (kubernetes-exteral-secrets, node-perf-dash)

In sum, I tried to converge on and document a few conventions:
- ensure `apps/{app}` matches the namespace declared in `infra/gcp/namespaces/ensure-namespaces.sh`
- ensure `apps/{app}/deploy.sh` exists and is referenced in `apps/{app}/README.md` - I opted for a copy pasted script, see below
- ensure `apps/{app}/OWNERS` has a `sig/foo` label in it
- ensure `apps/{app}/*.yaml` k8s resources have appropriate `labels: { app: {app} }` stanzas
- document these and other intended conventions in `apps/README.md`
- add known issues to `apps/README.md` and warning that conventions are subject to change as we iterate
- refresh `/README.md` to reference `apps` and link to https://github.com/kubernetes/k8s.io/issues/2150

The `deploy.sh` script:
- The script is copy-pasted to make it clear we could likely reduce down to a single parameterized script
- But I didn't reduce down yet, in case we find there are apps that don't fit this model and need to be customized as one-offs

A nice-to-have I did while I was here: sort the buckets used by gcsweb

### CODE REVIEW

**Comment by spiffxp** [comment]
/assign @thockin 
for post-merge FYI

**Comment by ameukam** [comment]
/lgtm
/hold
Remove hold when ready to deploy.

File: apps/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+Expect this to change as we iterate toward convergence on a standard that
+actually fits the apps we run today.
+
+## known issues
+
+- enforcement
+    - OWNER constraints enforcemed by humans
+    - no validation of yaml beyond yamllint
+    - `app:` convention enforcement by humands
```
Comment (jimdaga) [inline apps/README.md:34]: nit: spelling `humands`

Comment (spiffxp) [reply apps/README.md:34]: Good catch, I'll fix this in a followup

Comment (spiffxp) [reply apps/README.md:34]: Followup PR: https://github.com/kubernetes/k8s.io/pull/2314

**Pull request merged**